src/shared/CustomSelectDate.tsx

```diff
@@ -1,23 +1,31 @@
 import React from 'react';
-import "./CustomSelectDate.css"
+import './CustomSelectDate.css';
+import { DatePicker, Gapped } from '@skbkontur/react-ui';
 
-interface Props {
+interface PropsDateInput {
     readonly className?: string;
+    readonly date: string;
     readonly onChange?: (...rest: any) => void;
     readonly label: string;
     readonly classNameLabel?: string;
+    readonly height?: number;
+    readonly width?: number;
 }
 
-const CustomSelectDate = (props: Props) => {
+export const CustomSelectDate: React.FC<PropsDateInput> = ({
+                                                               className,
+                                                               classNameLabel,
+                                                               label,
+                                                               onChange,
+                                                               height,
+                                                               width = 320,
+                                                               date
+                                                           }) => {
     return (
-        <div className={props.className ?? 'custom_select_date'}>
-            <p className={props.classNameLabel ?? 'label_select_date'}>{props.label}</p>
-            <span className="datepicker-toggle">
-                <span className="datepicker-toggle-button"></span>
-                <input type="date" className="datepicker-input" onChange={props.onChange}/>
-            </span>
-        </div>
-    )
-}
-
-export default CustomSelectDate
\ No newline at end of file
+        <Gapped vertical gap={7}>
+            <p className={classNameLabel ?? 'custom-input_label'}>{label}</p>
+            <DatePicker style={{ width: width, height: height }} onValueChange={() => onChange} className={className}
```

Comment:
Возможно, имеет смысл присмотреться к такому форматтированию. Кстати, мне во Vue eslint даже порядок компонентов исправлял. У реакта такого нет?
```suggestion
            <DatePicker style={{ width: width, height: height }} 
                        onValueChange={() => onChange} 
                        className={className}
                        enableTodayLink value={date}
             />
```